### PR TITLE
Add GAlgebra 0.4.4

### PR DESCRIPTION
**Pull request by utensil**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/galgebra/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - python
+    - sympy
+
+test:
+  imports:
+    - galgebra
+
+about:
+  home: http://galgebra.readthedocs.io
+  license: BSD
```
Comment (chrisburr) [inline recipes/galgebra/meta.yaml]:
```suggestion
  license: BSD-3-Clause
```

File: recipes/galgebra/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  description: |
+    Symbolic Geometric Algebra/Calculus package for SymPy. BSD License.
+  doc_url: http://galgebra.readthedocs.io
+  dev_url: https://github.com/pygae/galgebra
+
+extra:
+  recipe-maintainers:
+    - utensil
+    - hugohadfield
```
Comment (scopatz) [inline recipes/galgebra/meta.yaml]: @hugohadfield are you OK with being maintainer

**Comment by isuruf** [comment]
@asmeurer, any thoughts about the naming? Is Alan not maintaining galgebra anymore?

**Comment by asmeurer** [comment]
Last I heard he was wanting to move it back into SymPy.

**Comment by asmeurer** [comment]
https://groups.google.com/forum/#!msg/sympy/ILzDtdNd83A/FZ-9ys6PBAAJ

File: recipes/galgebra/meta.yaml (hunk 2 above)
Comment (hugohadfield) [reply recipes/galgebra/meta.yaml]: Yes, I am

**Comment by hugohadfield** [comment]
Hi, @asmeurer and @isuruf we have tried a few times to get in contact with Alan with no response so far.. currently we have been working on just implementing the things that are immediately needed to keep the package alive, eg. Python 3, docs, autotesting etc.. We have a little github group to try and spread the maintenance burden of this stuff etc: https://github.com/pygae . If you know Alan personally it would be great if you could put us in touch..

**Comment by asmeurer** [comment]
The only way I know how to contact him is through his email and through his GitHub, which I'm assuming you already tried. 

What are your plans with the package? Do you plan to maintain it? Do you want to reintegrate it into SymPy?

**Comment by hugohadfield** [comment]
Yeah exactly, just keeping the package alive and up to date is the most important thing. I was discussing with @utensil yesterday using it as a reference implementation should we write a new symbolic GA package to address some of the performance problems of GAlgebra. The only real new features I could imagine adding are some minor syntactic additions to bring it more aligned with the other popular/good GA libraries. As for merging it back into SymPy I have no idea about the costs/benefits of that... @utensil do you have an opinion on that?

**Comment by utensil** [comment]
> https://groups.google.com/forum/#!msg/sympy/ILzDtdNd83A/FZ-9ys6PBAAJ

Oh, I'm not aware of this post. I guess Alan was trying to find a team to maintain the package when he's talking about merging back to SymPy. Actually now there is [Pythonic Geometric Algebra Enthusiasts](https://github.com/pygae) maintaining the community fork which supports Python 3, increases test coverage, set up CI and linters, maintains releases to PyPI, improves docs and has many bug fixes, please see [Changelog](https://galgebra.readthedocs.io/en/latest/changelog.html) for details. So this solves part of the problem.

As for merging back to SymPy, there're 3 roadblocks that I can see:

- SymPy did wish to separate less coupled modules from SymPy, as in sympy/sympy#8365 and I personally agree with this design decision
- GAlgebra went a long way after leaving SymPy, especially it changed a lot without conforming to SymPy requirements, it has many scattered pieces from different time points, and we haven't even reached the point to have a consistent doc for its actual interface and implementation
- GAlgebra heavily relies on `collect()` to separate the coefficients from basis but it's broken since SymPy 1.4, see https://github.com/pygae/galgebra/issues/31 , I'm still trying to figure out how to fix this and it might require work on both the GAlgebra side and the SymPy side

So IMHO, in the current stage, it's better for GAlgebra to be maintained in a separate repo and if merging back is what SymPy team wishes too, we can add a goal to conform to SymPy requirements but personally I might not be able to put much effort in it since I've seen a lot of other work in GAlgebra :

1. Performance: GAlgebra is extremely slow for certain coordinates and metrics, primarily caused by eagerness in the calculus stuff and simplification, so the main idea is laziness, so higher dimensions can be supported
2. Better support for FGA and CGA: it's already partially supported but I need to extensively verify them and some complications are expected
3. SymPy 1.4 support: I need to reimplement collect in SymPy for non-commutative symbols, that's quite some work, see https://github.com/pygae/galgebra/issues/31
4. Doc improvements: I want to write a true tutorial, weaving the examples together; also the docstring in Python code are mostly ill formated;
5. Examples fixing: examples in doc/python are not verified; some of them have serious correctness issues yet to be fixed.

**Comment by utensil** [comment]
I've just posted a reply to the post and it's will appear afer review.

As for our plan for GAlgebra, there's actually quite some work around GAlgebra :

- Added more examples in GAlgebra to enrich the potential use case scenarios
- [GAlgebra.jl](https://github.com/pygae/GAlgebra.jl): Julia interface to GAlgebra via PyCall and some syntax sugar experiment with it and many additional tests using syntax sugars that feel like reading/writing math: https://github.com/pygae/GAlgebra.jl/blob/master/test/runtests.jl#L262 
- [gajit](https://github.com/hugohadfield/gajit): A JIT compiler for geometric algebra in python which uses GAlgebra as well as Gaalop to symbolically optimizes geometric algebra algorithms, generates numpy code and JITs the resultant code, see https://rd.springer.com/chapter/10.1007/978-3-030-22514-8_50 

These work might not fully honor the potential of GAlgebra but we have certain interest in using and maintaining GAlgebra and we hope to do so beyond academic cycle (i.e. some GA libraries might be left unmaintained when the maintainers graduate or move to another area and time passes some other people might just start a new GA library instead of taking up the torch).

**Comment by asmeurer** [comment]
Whether or not to merge it back is entirely up to you. If you would rather maintain it separately, then go ahead and do so. The primary benefits of merging it back are to you, and the downsides are to us. The benefit is that it stays maintained with the SymPy codebase. So for instance, no PR in SymPy would be merged that breaks a galgebra test, because it would fail the CI. The downside is more maintenance for us. We're also happy to update any documentation in SymPy to point to this repo (although it would be good to hear back from Alan first). 

The main reason we moved it out was that it was somewhat isolated from the rest of SymPy, both in code and community. Few developers outside of Alan were able to maintain it. The code didn't actually use the core of SymPy, and no other part of SymPy used it (I'm sure this second part would continue to be true, unless there are use-cases of galgebra in other things in SymPy that you know of). So it ended up being effectively a separate module, even though it lived in the SymPy repo. As it now uses the core of SymPy, as I understand it, this isn't really the case any more. 

> I've just posted a reply to the post and it's will appear afer review.

I've approved your post.

**Comment by utensil** [comment]
> Whether or not to merge it back is entirely up to you. If you would rather maintain it separately, then go ahead and do so. The primary benefits of merging it back are to you, and the downsides are to us.

Yeah, thanks for the clarification.

> The main reason we moved it out was that it was somewhat isolated from the rest of SymPy, both in code and community. .......  So it ended up being effectively a separate module, even though it lived in the SymPy repo. 

Yeah, this is exactly the case and also why I agree with the decision.

**Pull request merged**